Libraries Tasmania.js

```diff
@@ -0,0 +1,781 @@
+	// This makes it difficult to get at them reliably.
+	// Because there are so many variations that could break things, processDocs calls item.done() on error.
+	if (digitalUrls.length > 0) {
+		ZU.processDocuments(digitalUrls, function (digitalDoc) {
+			var mimeType = "image/jpeg";
+			// Find the download link
+			var downloadLink = digitalDoc.querySelector("a[title='Download'], a[id='downloadLink']");
+			// Sometimes the download link opens a modal, if so get the link from the modal code
+			if (downloadLink && !downloadLink.href.includes("/download/")) {
+				downloadLink = digitalDoc.querySelector("div.downloadDialog a");
+			}
+			if (downloadLink) {
+				// Z.debug(downloadLink.href);
+				// If the page is using a PDF viewer, set mime type to pdf
+				// Otherwise use default jpeg mimetype
+				let viewer = digitalDoc.querySelector("#viewer");
+				if (viewer && viewer.className.startsWith("pdf")) {
+					mimeType = "application/pdf";
+				}
+				// Zotero.debug(digitalDoc.location.href);
+				// Add file as attachment
+				item.attachments.push({
+					title: 'Libraries Tasmania digital item: ' + digitalLabels[digitalDoc.location.href],
+					mimeType: mimeType,
+					url: downloadLink.href
+				});
+			}
+		},
+		// On complete
+		function () {
+			item.complete();
+		},
+		// On error
+		function (e) {
+			Z.debug(e);
+			item.complete();
+		});
+	}
+	else {
+		item.complete();
+	}
+}
+
+function cleanText(text) {
+	// Removes unnecessary stuff from catalogue strings
+	var replace = ["[", "]", "<", ">"];
+	text = ZU.trim(text);
+	for (let i = 0; i < replace.length; ++i) {
+		text = text.replace(replace[i], "");
+	}
+	text = text.replace(/\.$/, "");
+	return text;
+}
+
+function scrape(doc, url) {
+	// Catalogue search type
+	var catType = url.match(/en_AU\/(library|tas|names|all)\/search/)[1];
+	// The 'all' search type can include anything.
+	// In this case URL checks should identify archives and names.
+	// Including both checks to try and catch as much as possible.
+	if (catType == "tas" || url.includes("ARCHIVES_")) {
+		scrapeArchives(doc, url);
+	}
+	else if (catType == "names" || url.includes("NAME_INDEXES")) {
+		scrapeNames(doc, url);
+	}
+	else {
+		scrapeLibrary(doc, url);
+	}
+}
+
+function scrapeLibrary(doc, url) {
+	// Format of item
+	var format = detectWeb(doc, url);
+	// If there's a 'Library View' tab we can use MARC
+	if (doc.querySelector("div#marc-tab0")) {
+		// call MARC translator
+		var translator = Zotero.loadTranslator("import");
+		translator.setTranslator("a6ee60df-1ddc-4aae-bb25-45e0537be973");
+		translator.getTranslatorObject(function (marc) {
+			var record = new marc.record();
+			var item = new Zotero.Item();
+			var fieldTag, indicators, fieldContent;
+			for (let row of doc.querySelectorAll("div#marc-tab0 table tr")) {
+				let fields = row.querySelectorAll("td");
+				fieldTag = ZU.trim(fields[0].textContent);
```

Comment:
`ZU.trim` can be replaced by `String#trim()` here and everywhere else it's used:

```suggestion
				fieldTag = fields[0].textContent.trim();
```

We eventually need to remove it - it was just a shim before browsers supported that built-in function.